src/main/java/com/example/TicketDAO.java

```diff
@@ -1,54 +1,59 @@
 package com.example;
 
-import java.sql.Connection;
-import java.sql.PreparedStatement;
-import java.sql.ResultSet;
-import java.sql.SQLException;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.EntityManagerFactory;
+import jakarta.persistence.Persistence;
+
 
 public class TicketDAO {
-    private static final String SAVE_TICKET_SQL = "INSERT INTO Ticket (user_id, ticket_type) VALUES (?, ?::ticket_type)";
-    private static final String FETCH_TICKET_SQL = "SELECT * FROM Ticket WHERE id = ?";
-    private static final String UPDATE_TICKET_SQL = "UPDATE Ticket SET ticket_type = ?::ticket_type WHERE id = ?";
+    private final EntityManagerFactory emf = Persistence.createEntityManagerFactory("my_persistence");
 
     // Method to save a ticket to the database
-    public void saveTicket(Ticket ticket) throws SQLException {
-        try (Connection conn = DatabaseConnection.getConnection();
-             PreparedStatement stmt = conn.prepareStatement(SAVE_TICKET_SQL)) {
-
-            stmt.setInt(1, ticket.getUserId());
-            stmt.setString(2, ticket.getTicketType().name());  // Make sure to match enum name
-
-            stmt.executeUpdate();
+    public void saveTicket(Ticket ticket) {
+        //EntityManager em = emf.createEntityManager();
+
+        try(EntityManager em = emf.createEntityManager()) {
+            em.getTransaction().begin();
+            em.persist(ticket);
+            em.getTransaction().commit();
+            System.out.println("Saved new ticket for user ID: " + ticket.getUserId() + " with type: " + ticket.getTicketType());
+        } catch (Exception e) {
+            {
+                System.err.println("Error saving ticket: " + e.getMessage());
+                e.printStackTrace();
+            }
```

Comment:
Why do we need this block?
I think **{** (21 line) and **}** (24 line) are redundant.